src/software/new_geom/util/closest_point_test.cpp

```diff
@@ -46,3 +46,19 @@ TEST(ClosestPointTest, point_far_from_line)
     EXPECT_EQ(closestPointOnLine(p, l), expected);
     EXPECT_EQ(closestPointOnLine(l, p), expected);
 }
+
+TEST(GeomUtilTest, test_closest_lineseg_point)
+{
+    Segment seg(Point{-1, 1}, Point{1, 1});
+
+    EXPECT_TRUE((closestPointOnSeg(Point(0, 2), seg) - Point(0, 1)).length() < 0.00001);
+    EXPECT_TRUE((closestPointOnSeg(Point(-2, 1.5), seg) - Point(-1, 1)).length() <
+                0.00001);
+
+    seg = Segment(Point{-2, 1}, Point{1, 2});
+
+    EXPECT_TRUE((closestPointOnSeg(Point(1, 0), seg) - Point(0.4, 1.8)).length() <
+                0.00001);
+    EXPECT_TRUE((closestPointOnSeg(Point(-1.4, 1.2), seg) - Point(-1.4, 1.2)).length() <
+                0.00001);
+}
```

Comment:
Looks like there are two test cases here, what are they testing? Ideally we make them separate tests with descriptive names. It would also be good to add tests for a point somewhere near the middle of the segment, as well as at the endpoints of the segment?

Reply:
Split into 5 cases now:
- Point is on seg
- Point closest is the middle
- Point closest is the start
- Point closest is the end
- Point closest is anywhere else (intermediate)

Reply:
Nice! Looks good.